### PR TITLE
Add room intro warning when e2ee is not enabled

### PR DESCRIPTION
**Pull request by gsouquet**
Fixes vector-im/element-web#15675

Screen Shot 2021-04-27 at 14 43 19

This message is triggered as part of `RoomIntro` (it will be more visible once #5887 lands) and only is displayed if:
* It is a private room
* The server is configured so that private conversation should be encrypted by default
* The current room is not encrypted

@matrix-org/design keen to hear your thoughts on this as I could not find any "inline" alert to explain a potentially dangerous action. Everything happens in a modal which seemed too intrusive for this scenario in my opinion.
More than happy to tweak this to make sure that it provides the most consistent experience possible

### CODE REVIEW

**Review by jryans — APPROVED** [review]
Seems fine code wise, though I suspect Design will want to tweak this one way or another.

**Comment by nadonomy** [comment]
@gsouquet after exploring some options I ended up arriving at this:

https://www.figma.com/file/ccOtQyqevLDXE6UktupgN3/E2E-Room-Warnings?node-id=4%3A3188

It made sense to me to use the same contextual warnings we do to communicate encryption state, and by persisting it's place in the timeline we can afford to be more verbose in the copy, which let's us (1) build trust & confidence (2) not alarm the user & (3) offer help on how to remedy.

Let me know if you have any thoughts!

**Comment by gsouquet** [comment]
Here is a screenshot of the implementation on Element Web (cc @nadonomy)

Screen Shot 2021-06-10 at 11 15 29

**Comment by nadonomy** [comment]
@gsouquet thanks! This is looking great.

I have some copy requests actually with the benefit of fresh eyes. 😀 Able to make the language more terse and front-load sentiment clearer.

Would you mind updating the title to:

> End-to-end encryption isn't enabled

And then the body copy to:

> Your private messages are normally encrypted, but this room isn't. Usually this is due to an unsupported device or method being used, like email invites. Enable encryption in Settings.

File: src/components/views/rooms/NewRoomIntro.tsx [hunk 1]
```diff
@@ -166,7 +177,29 @@ const NewRoomIntro = () => {
...
+        dis.dispatch({
+            action: "open_room_settings",
+            initial_tab_id: ROOM_SECURITY_TAB,
+        });
+    }
+
+    const sub2 = _t("Your private messages are normally encrypted, but this room isn't. "+
+    "Usually this is due to an unsupported device or method being used, " +
+    "like email invites. <a>Enable encryption in settings.</a>", {},
+    { a: sub => <a onClick={openRoomSettings} href="#">{sub}</a> });
```
Comment (jryans) [inline src/components/views/rooms/NewRoomIntro.tsx:191]: My brain expects some indentation for a multi-line statement like this... perhaps something like:

```suggestion
    const sub2 = _t(
        "Your private messages are normally encrypted, but this room isn't. "+
        "Usually this is due to an unsupported device or method being used, " +
        "like email invites. <a>Enable encryption in settings.</a>", {},
        { a: sub => <a onClick={openRoomSettings} href="#">{sub}</a> },
    );
```

**Review by jryans — APPROVED** [review]

File: src/components/views/rooms/NewRoomIntro.tsx [hunk 2]
```diff
@@ -31,6 +31,17 @@ import dis from "../../../dispatcher/dispatcher";
 import SpaceStore from "../../../stores/SpaceStore";
 import {showSpaceInvite} from "../../../utils/space";
 
+import { privateShouldBeEncrypted } from "../../../createRoom";
+
+import EventTileBubble from "../messages/EventTileBubble";
+import { ROOM_SECURITY_TAB } from "../dialogs/RoomSettingsDialog";
+
+function hasExpectedEncryptionSettings(room): boolean {
+    const isEncrypted: boolean = room._client?.isRoomEncrypted(room.roomId);
```
Comment (t3chguy) [inline src/components/views/rooms/NewRoomIntro.tsx:40]: Could we please avoid relying on internal/private fields, it makes Typescript conversions that bit riskier. Not so bad here as TS caught it but if it was in JS 💥

**Pull request merged**